applicationset/controllers/applicationset_controller.go

```diff
@@ -1424,4 +1429,63 @@ func shouldRequeueApplicationSet(appOld *argov1alpha1.Application, appNew *argov
 	return false
 }
 
+func (r *ApplicationSetReconciler) getStrategySpecFromRef(ctx context.Context, ref *argov1alpha1.SyncStrategyRef, namespace string) (*argov1alpha1.SyncStrategySpec, error) {
+	if ref == nil {
+		return nil, nil
+	}
+
+	if ref.Kind != application.ClusterSyncStrategyKind && ref.Kind != application.SyncStrategyKind {
+		return nil, fmt.Errorf("unsupported kind for strategy: %s", ref.Kind)
+	}
+
+	var spec *argov1alpha1.SyncStrategySpec
+	if ref.Kind == application.ClusterSyncStrategyKind {
+		css := &argov1alpha1.ClusterSyncStrategy{}
+		err := r.Get(
```

Comment:
You might have to update the default argocd ClusterRole/Role to explicitly add this. It makes it easier for people with custom roles other than *